chainer/optimizers/adam.py

```diff
@@ -99,18 +110,30 @@ def update_core_cpu(self, param):
                 'eps of Adam optimizer is too small for {} ({})'.format(
                     grad.dtype.name, hp.eps))
         m, v = self.state['m'], self.state['v']
-
-        m += (1 - hp.beta1) * (grad - m)
-        v += (1 - hp.beta2) * (grad * grad - v)
-
-        if hp.amsgrad:
-            vhat = self.state['vhat']
-            numpy.maximum(vhat, v, out=vhat)
+        if (isinstance(m, intel64.mdarray) and
+            isinstance(v, intel64.mdarray)):
+            m.inplace_axpby(1.0, 1.0 - hp.beta1, grad - m)
+            v.inplace_axpby(1.0, 1.0 - hp.beta2, grad*grad - v)
+            if hp.amsgrad:
```

Comment:
Let's align the code style with the non-ideep case.

```py
```
if hp.amsgrad:
    vhat = self.state['vhat']
    numpy.maximum(vhat, v, out=vhat)
else:
    vhat = v
param.data.inplace_axpby(1.0 - hp.weight_decay_rate, -hp.eta,
                         self.lr * m / (numpy.sqrt(vhat) + hp.eps))
```